walkthrough/step05.md

```diff
@@ -0,0 +1,190 @@
+# Step05 - Auth middleware
+
+Now that we have our user passwords secured using bcrypt, and our cookies encoded with JWT, we can safley say that we took a step forward into having a secured website.
+
+There are many more steps that we can take to secure our website properly(links would be available at the bottom), but for now this is a good basis.
+
+<img src="" alt="" styles="text-align:center;" />
+
+**insert security related image here**
+
+
+Right now the only place where we check if the user is logged in and valid is in the homepage, what if we wanted to check if he is logged in 10 different other routes ?
+So do we write the same code multple time ? No cause then it is not considered to be DRY code (Don't Repeat Yourself)
```

Comment:
```suggestion
Should we write the same code for different routes? 
**No** because then it is not DRY (Don't Repeat Yourself) code, remember that you should be encapsulating code when you can :sparkles: 
```